baybe/campaign.py

```diff
@@ -342,3 +336,10 @@ def _add_version(dict_: dict) -> dict:
     Campaign, lambda x: _add_version(unstructure_hook(x))
 )
 converter.register_structure_hook(Campaign, structure_hook)
+
+
+# Converter for config validation
```

Comment:
I am confused by this: According to your comment, it seems like this is related to the issue that prevented us from moving to 3.12. However, isn't the code now just at another place within the file? Am I missing something here or could you maybe elaborate a bit on this?

Reply:
Yes, exactly. But because it is at another place, i.e. *after* the hooks have been added to the regular converter, these hooks are now also part of the config converter because the copying happens later. And therefore, the validation converter now also has an explicit hook registered for `Campaign` which circumvents (emphasis is "circumvents", not "solves") the problem. 

The root cause is in fact a bug in Python 3.12 itself, which seems to have a problem in `functools.singledispatch`.  I've opened a corresponding python ticket yesterday evening. You'll find the link at the end of the other issue I mentioned

Reply:
Can I resolve?

Reply:
No, because, I resolve this 😎  Thanks for the explanation